mava/wrappers/jumanji.py

```diff
@@ -117,3 +118,74 @@ def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
 
         # Aggregate the list of individual rewards and use a single team_reward.
         return self.aggregate_rewards(timestep, modified_observation)
+    
+
+class ConnectorWrapper(MultiAgentWrapper):
+    """Multi-agent wrapper for the MA Connector environment."""
+
+    def __init__(self, env: MaConnector):
+        super().__init__(env)
+
+    def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
+        """Modify the timestep for the Connector environment."""
+
+        def create_agents_view(grid: chex.Array) -> chex.Array:
+            positions = jnp.where(grid % 3 == 2, True, False)   
+            targets = jnp.where((grid % 3 == 0) & (grid != 0), True, False)
+            paths = jnp.where(grid % 3 == 1, True, False)
+            my_position = jnp.where(grid == 2, True, False)
+            my_target = jnp.where(grid == 3, True, False)
+            agents_view = jnp.stack((positions, targets, paths, my_position, my_target), -1)
+            return agents_view
+        
+        def create_global_state(grid: chex.Array) -> chex.Array:
+            positions = jnp.where(grid % 3 == 2, True, False)   
+            targets = jnp.where((grid % 3 == 0) & (grid[0] != 0), True, False)
+            paths = jnp.where(grid % 3 == 1, True, False)
+            global_state = jnp.stack((positions, targets, paths), -1)
+            return global_state
+        
+        observation = ObservationGlobalState(
+            global_state = create_agents_view(timestep.observation.grid),
+            agents_view= create_global_state(timestep.observation.grid),
+            action_mask=timestep.observation.action_mask,
+            step_count=jnp.repeat(timestep.observation.step_count, self._num_agents),
+        ) 
+        return timestep.replace(observation=observation)
+    
+    
+    def observation_spec(self) -> specs.Spec[Observation]:
+        """Specification of the observation of the environment."""
+        step_count = specs.BoundedArray(
+            (self._num_agents,),
+            jnp.int32,
+            [0] * self._num_agents,
+            [self.time_limit] * self._num_agents,
```

Comment:
Use jnp functions wherever possible

```suggestion
            jnp.zeros(self._num_agents, dtype=int),
            jnp.repeat(self.time_limit, self._num_agents),
```

Reply:
That code is actually copied from the generic wrapper. I guess we should make another PR for that too?

Reply:
That's quite related to this PR, so you can squeeze it in here if it's easier :smile: